File: token_type.py
from enum import Enum
from typing import List

SINGLE_CHAR_TOKENS: List[str] = [
    'LEFT_PAREN', 'RIGHT_PAREN', 'LEFT_BRACE', 'RIGHT_BRACE',
    'COMMA', 'DOT', 'MINUS', 'PLUS', 'SEMICOLON', 'SLASH', 'STAR',
]

ONE_OR_TWO_CHAR_TOKENS: List[str] = [
    'BANG', 'BANG_EQUAL',
    'EQUAL', 'EQUAL_EQUAL',
    'GREATER', 'GREATER_EQUAL',
    'LESS', 'LESS_EQUAL',
]

LITERALS: List[str] = ['IDENTIFIER', 'STRING', 'NUMBER']

KEYWORDS: List[str] = [
    'AND', 'CLASS', 'ELSE', 'FALSE', 'FUN', 'FOR', 'IF', 'NIL', 'OR',
    'PRINT', 'RETURN', 'SUPER', 'THIS', 'TRUE', 'VAR', 'WHILE',
    'EOF'
]

ALL_TOKENS: List[str] = SINGLE_CHAR_TOKENS + ONE_OR_TWO_CHAR_TOKENS + LITERALS + KEYWORDS

TOKEN_TYPE: Enum = Enum('TokenType', ALL_TOKENS)

File: tokenizer/tokentype.go
package tokenizer

// TokenType type
type TokenType int

// Token types
const (
	// Single Character Tokens
	LEFTPAREN TokenType = iota + 1
	RIGHTPAREN
	LEFTBRACE
	RIGHTBRACE
	COMMA
	DOT
	MINUS
	PLUS
	SEMICOLON
	SLASH
	STAR

	// One or two character tokens
	BANG
	BANGEQUAL
	EQUAL
	EQUALEQUAL
	GREATER
	GREATEREQUAL
	LESS
	LESSEQUAL

	// Literals
	IDENTIFIER
	STRING
	NUMBER

	// Keywords
	AND
	CLASS
	ELSE
	FALSE
	FUN
	FOR
	IF
	NIL
	OR
	PRINT
	RETURN
	SUPER
	THIS
	TRUE
	VAR
	WHILE

	// End of File
	EOF
)
